### ISSUE TITLE
Caddyfile: add ability to set `strict_sni_host` to false

### ISSUE THREAD

**Issue by llchan** [caddyserver/caddy#4488]
The current parsing only allows for `true` or `nil`, which basically means the client-auth-triggered logic that turns on `strict_sni_host` is not configurable.  We should allow `strict_sni_host = false` at the global level and maybe add something inside the client auth block to disable this requirement per-client-auth.

**Comment by francislavoie** [comment]
What's your usecase for this exactly? Why do you need to enable client auth but have strict SNI off?

See in the code, there's a note as to why it's implicitly enabled.

https://github.com/caddyserver/caddy/blob/5333c3528bd2badf1676efcced322d151e3706c8/modules/caddyhttp/server.go#L288-L309

**Comment by llchan** [comment]
The use case was to shift the TLS handling of a Hashicorp Nomad server out to a sidecar container, since Nomad currently only has a single HTTP(S) listener and it can only have TLS on or off.  We had a use case for multiple ports, with/without TLS, and we were trying to use Caddy as the intermediary.  To emulate the Nomad TLS client cert behavior we'd need checking but allow the SNI to be set differently from the actual host header.  If this is a WONTFIX that's fine, we're using haproxy now instead so we're not blocked on this anymore.

**Comment by mholt** [comment]
>  I think this should be configurable

Well, this is configurable. The bool is nullable, so you can set it to false in addition to unset. See the above code block.

As you're not actually using Caddy, we don't have a use case to make any changes, so closing.
